### PR TITLE
added solution for LC problems 240 and 244

### LINKED ISSUE #3345: [Feature Request]: Adding missing solutions for Leetcode problems 240 and 244.
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

I propose adding documented solutions for the following LeetCode problems to the project repository:

Problem 240: Search a 2D Matrix II
Problem 244: Shortest Word Distance II
This involves providing clear, commented code solutions along with explanations of the approaches used for each problem.

### Use Case

Use Case for Problem 240 (Search a 2D Matrix II):

1.Efficiently searching for a target value in a 2D matrix where each row and column are sorted.
2.Implementing algorithms for quick data retrieval in matrix-like data structures, useful in applications like geographical mapping and spreadsheets.
Use Case for Problem 244 (Shortest Word Distance II):

1.Finding the shortest distance between two words in a list of words, which is essential for text processing tasks.
2.Enhancing algorithms for text analysis, natural language processing, and information retrieval by optimizing word distance calculations.

### Benefits

Practical Solutions: Provides practical implementations for efficient data search in 2D matrices and optimized text processing.
Educational Resource: Enhances project documentation with detailed examples of search algorithms, matrix manipulation, and text analysis techniques.
Community Engagement: Encourages collaborative learning and contributions, fostering a stronger developer community around algorithmic problem-solving.

### Add ScreenShots

_No response_

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by aaryaanshh**
## Fixed Issue 

Fix #3345 

## Related Issue

None

## Description

This pull request includes comprehensive documentation updates for the following LeetCode problems:
- **Problem 240: Search a 2D Matrix II**
- **Problem 244: Shortest Word Distance II**

The documentation consists of problem descriptions, examples, constraints, approaches to solve the problems, and complexity analyses.

## Type of PR

- [ ] Bug fix
- [ ] Feature enhancement
- [x] Documentation update
- [ ] Security enhancement
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)

Not applicable.

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have followed the code style guidelines of this project.
- [x] I have checked for any existing open issues that my pull request may address.
- [x] I have ensured that my changes do not break any existing functionality.
- [x] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently.
- [x] I have read the resources for guidance listed below.
- [x] I have followed security best practices in my code changes.

## Additional Context

This documentation aims to provide users with clear and structured approaches to solve the problems using different programming languages and techniques, enhancing their understanding and problem-solving skills.

## Resources for Guidance

Please read the following resources before submitting your contribution:

- [ ] [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [ ] [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [ ] [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [ ] [Markdown Guide](https://www.markdownguide.org/)
- [ ] [MDX Documentation](https://mdxjs.com/docs/)
- [ ] [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- [ ] [Tabs: Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [ ] [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [ ] [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex) 

## Contribution Guidelines

Thank you for considering contributing to our project! To ensure smooth collaboration and effective contribution management, please adhere to the following guidelines:

### Issue Creation

1. **Limit on Issues:**
   - Each contributor is allowed to create a maximum of **4 issues per day**. This helps us manage and address issues efficiently.

### Contribution Levels

2. **Basic Contributions:**
   - This project is primarily focused on documentation. Most of the setup has been completed, so contributors will generally need to work on basic code tasks, such as writing tests.
   - For these tasks, issues will be assigned the **level1** label.

3. **Acknowledging Hard Work:**
   - If a contributor puts in significant effort on a task, the issue will be upgraded to **level2**. This is our way of recognizing and appreciating extra effort.

4. **Feature Additions and Component Work:**
   - Contributors working on new features or components using JSX/TSX will be assigned a level based on the complexity and quality of their work.
   - The more complex and valuable the contribution, the higher the level assigned.

### Level Definitions

- **Level1:**
  - Tasks are straightforward, such as fixing minor bugs, writing tests, or making simple documentation updates.
- **Level2:**
  - Tasks require more effort, such as addressing complex bugs, improving existing features, or making substantial documentation improvements.
- **Level3:**
  - Tasks are highly complex and involve significant new feature development, major refactoring, or extensive contributions to the project’s core components.

We look forward to your contributions and appreciate your effort in helping us improve the project!

### CODE REVIEW

**Comment by aaryaanshh** [comment]
@Ajay-Dhangar Sir kindly review it and let me know if any corrections are needed

And sir how to create more than one PR at a time? You can tell me in short, then i will research further on it

File: dsa-solutions/lc-solutions/0200-0299/0244-shortest-word-distance-II.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    WordInfo* word2Info = NULL;
+    
+    for (int i = 0; i < obj->size; i++) {
+        if (strcmp(word1, word2) == 0 && strcmp(word1, word2) == 0) {
+            word1Info = &obj->wordMap[i];
+            word2Info = word1Info;
+            break;
+        }
+        if (strcmp(word1, obj->wordMap[i].name) == 0) {
+             Returns Solution  I O also have no   even ActionController know
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0200-0299/0244-shortest-word-distance-II.md:197]: Please ensure that you complete your code, and I strongly recommend that you follow our structure at https://github.com/orgs/CodeHarborHub/discussions/3369.

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
Please update your code based on the suggestions provided. Ensure that you complete your code and follow our structure as outlined in [this discussion](https://github.com/orgs/CodeHarborHub/discussions/3369).

**Comment by Ajay-Dhangar** [comment]
@aaryaanshh, please complete your pending pull requests.

**Comment by aaryaanshh** [comment]
> @aaryaanshh, please complete your pending pull requests.
Yes sir, will do as per the changes asked. Pls give me some time

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**